### PR TITLE
UNDERTOW-1773 to avoid potential leak, make sure the iteration continue in case of exception.

### PR DESCRIPTION
**Pull request by soul2zimate**
issue: https://issues.redhat.com/browse/UNDERTOW-1773
UNDERTOW-1773 to avoid potential leak, make sure the iteration continue in case of exception.

### CODE REVIEW

File: servlet/src/main/java/io/undertow/servlet/core/ApplicationListeners.java [hunk 1]
```diff
@@ -325,7 +326,12 @@ public void sessionDestroyed(final HttpSession session) {
         final HttpSessionEvent sre = new HttpSessionEvent(session);
         for (int i = httpSessionListeners.length - 1; i >= 0; --i) {
             ManagedListener listener = httpSessionListeners[i];
-            this.<HttpSessionListener>get(listener).sessionDestroyed(sre);
+            try {
+                this.<HttpSessionListener> get(listener).sessionDestroyed(sre);
+            } catch (Exception e) {
+                // UNDERTOW-1773 to avoid leak, make sure the iteration continue in case of exception.
+                UndertowLogger.ROOT_LOGGER.debugf(e, "Exception happens in sessionDestroyed for session %s ", session);
```
Comment (stuartwdouglas) [inline servlet/src/main/java/io/undertow/servlet/core/ApplicationListeners.java:333]: This should really be at error level.

Comment (soul2zimate) [reply servlet/src/main/java/io/undertow/servlet/core/ApplicationListeners.java]: Thank you, I have updated the log level to ERROR

**Comment by fl4via** [comment]
As has been discussed in the JBEAP Jira, we should apply the same fix to other listener notifications as well.

**Comment by soul2zimate** [comment]
> As has been discussed in the JBEAP Jira, we should apply the same fix to other listener notifications as well.

OK, I will apply them to all other listener notifications.

**Comment by soul2zimate** [comment]
I have updated the commit to cover all other methods with listener invocations

**Review by fl4via — APPROVED** [review]

**Pull request merged**